homeassistant/components/shelly/switch.py

```diff
@@ -265,13 +243,22 @@ async def async_added_to_hass(self) -> None:
             self.last_state = last_state
 
 
-class BlockRelaySwitch(ShellyBlockEntity, SwitchEntity):
+class BlockRelaySwitch(ShellyBlockAttributeEntity, SwitchEntity):
     """Entity that controls a relay on Block based Shelly devices."""
 
-    def __init__(self, coordinator: ShellyBlockCoordinator, block: Block) -> None:
+    entity_description: BlockSwitchDescription
+
+    def __init__(
+        self,
+        coordinator: ShellyBlockCoordinator,
+        block: Block,
+        attribute: str,
+        description: BlockSwitchDescription,
+    ) -> None:
         """Initialize relay switch."""
-        super().__init__(coordinator, block)
+        super().__init__(coordinator, block, attribute, description)
         self.control_result: dict[str, Any] | None = None
+        self._attr_unique_id: str = f"{coordinator.mac}-{block.description}"
```

Comment:
Since we are recalculating the unique ids here, would you please add tests for entities with `sensor|motionActive` and `relay|output` in a separate PR so we can merge that, and than rebase this PR to verify we have a 1:1 unique id with the new code?

Reply:
Aren't those  enough ?

- <https://github.com/home-assistant/core/blob/dev/tests/components/shelly/test_switch.py#L40>
- <https://github.com/home-assistant/core/blob/dev/tests/components/shelly/test_switch.py#L64>

The "recalculation" is due to the switch from `ShellyBlockEntity` to `ShellyBlockAttributeEntity` but there is no logic change in the end.

Reply:
I was looking for a test that setup the entities, fetched the unique id from the registry and asserted it was in the expected format for `"relay|output"` and `"sensor|motionActive"`

Reply:
https://github.com/home-assistant/core/pull/139778